acme_test.go

```diff
@@ -155,6 +175,66 @@ func TestAuthorize(t *testing.T) {
 	}
 }
 
+func TestAcceptChallenge(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "HEAD" {
+			w.Header().Set("replay-nonce", "test-nonce")
+			return
+		}
+		if r.Method != "POST" {
+			t.Errorf("r.Method = %q; want POST", r.Method)
+		}
+
+		j := struct {
+			Resource string
+			Type     string
+			Auth     string `json:"keyAuthorization"`
+		}{}
+		decodeJWSRequest(t, &j, r)
```

Comment:
I mean, I'm fine with it either way, but thought you should know. It's more idiomatic.